bash-preexec.sh

```diff
@@ -250,10 +250,8 @@ __bp_preexec_invoke_exec() {
     fi
 
     local this_command
-    this_command=$(
-        export LC_ALL=C
-        HISTTIMEFORMAT='' builtin history 1 | sed '1 s/^ *[0-9][0-9]*[* ] //'
-    )
+    this_command=$(LC_ALL=C HISTTIMEFORMAT='' builtin history 1)
+    this_command="${this_command# *[[:digit:]]*[* ] }"
```

Comment:
The pattern in `sed` is a regular expression so cannot be directly used in `${var#pat}`, where `pat` is a globbing pattern. For example, the above fails when the history size is larger or equal to 10000 because the line would be missing the beginning space like `10000  history 1`. The ` *` at the beginning of the pattern requires at least one space and an arbitrary (shortest) string.

Reply:
I actually just realized that (in the shower!) and was coming back to my computer to fix it when I saw your message. 🙂